src/Scripting/Core/ScriptBuilder.cs

```diff
@@ -73,20 +73,13 @@ public int GenerateSubmissionId(out string assemblyName, out string typeName)
             {
                 // get compilation diagnostics first.
                 diagnostics.AddRange(compilation.GetParseDiagnostics());
-                if (diagnostics.HasAnyErrors())
-                {
-                    CompilationError(diagnostics, compiler.DiagnosticFormatter);
-                }
-
+                ThrowIfAnyCompilationErrors(diagnostics, compiler.DiagnosticFormatter);
                 diagnostics.Clear();
```

Comment:
I can imagine only reporting warnings if there are *no* errors, but generally speaking warnings should be reported unless the user suppresses them.